gix-config/src/file/access/raw.rs

```diff
@@ -434,43 +401,35 @@ impl<'event> File<'event> {
     /// # use bstr::BStr;
     /// # use std::convert::TryFrom;
     /// # let mut git_config = gix_config::File::try_from("[core]a=b").unwrap();
-    /// let prev = git_config.set_raw_value("core", None, "a", "e")?;
-    /// git_config.set_raw_value("core", None, "b", "f")?;
+    /// let prev = git_config.set_raw_value("core.a", "e")?;
+    /// git_config.set_raw_value("core.b", "f")?;
     /// assert_eq!(prev.expect("present").as_ref(), "b");
-    /// assert_eq!(git_config.raw_value("core", None, "a")?, Cow::<BStr>::Borrowed("e".into()));
-    /// assert_eq!(git_config.raw_value("core", None, "b")?, Cow::<BStr>::Borrowed("f".into()));
+    /// assert_eq!(git_config.raw_value("core.a")?, Cow::<BStr>::Borrowed("e".into()));
+    /// assert_eq!(git_config.raw_value("core.b")?, Cow::<BStr>::Borrowed("f".into()));
     /// # Ok::<(), Box<dyn std::error::Error>>(())
     /// ```
-    pub fn set_raw_value<'b, Key, E>(
+    pub fn set_raw_value<'b, E>(
         &mut self,
-        section_name: impl AsRef<str>,
-        subsection_name: Option<&BStr>,
-        key: Key,
+        key: impl Key<'b>,
         new_value: impl Into<&'b BStr>,
     ) -> Result<Option<Cow<'event, BStr>>, crate::file::set_raw_value::Error>
     where
-        Key: TryInto<section::Key<'event>, Error = E>,
         section::key::Error: From<E>,
     {
-        self.set_raw_value_filter(section_name, subsection_name, key, new_value, &mut |_| true)
+        self.set_raw_value_filter(key, new_value, &mut |_| true)
     }
 
     /// Similar to [`set_raw_value()`][Self::set_raw_value()], but only sets existing values in sections matching
     /// `filter`, creating a new section otherwise.
-    pub fn set_raw_value_filter<'b, Key, E>(
+    pub fn set_raw_value_filter<'b>(
         &mut self,
-        section_name: impl AsRef<str>,
-        subsection_name: Option<&BStr>,
-        key: Key,
+        key: impl Key<'b>,
         new_value: impl Into<&'b BStr>,
         filter: &mut MetadataFilter,
-    ) -> Result<Option<Cow<'event, BStr>>, crate::file::set_raw_value::Error>
-    where
-        Key: TryInto<section::Key<'event>, Error = E>,
-        section::key::Error: From<E>,
-    {
-        let mut section = self.section_mut_or_create_new_filter(section_name, subsection_name, filter)?;
-        Ok(section.set(key.try_into().map_err(section::key::Error::from)?, new_value.into()))
+    ) -> Result<Option<Cow<'event, BStr>>, crate::file::set_raw_value::Error> {
+        let mut section = self.section_mut_or_create_new_filter(key.section_name(), key.subsection_name(), filter)?;
+        let key = key.name().to_owned().try_into().map_err(section::key::Error::from)?; // TODO Borrow, plz
```

Comment:
This `to_owned()` is not optimal, but it is not obvious to me how to formulate the required lifetimes.